### ISSUE TITLE
We need to decide how to merge new content into our templates

### ISSUE THREAD

**Issue by wslack** [18F/federalist#880]
Existing template (uswds-jekyll): https://federalist-proxy.app.cloud.gov/site/18f/federalist-uswds-template/

New work (OES site adapted from uswds-jekyll): https://federalist.fr.cloud.gov/preview/gsa-oes/office-of-evaluation-sciences/update-uswds/

The differences between the OES template and the uswds-jekyll template are: 
- basic navigation (nested beside the logo not below), 
- logo image and text 
- new homepage sections (billboard/call out, contact)
- updated hero treatment
- alternate footer 
- color block titles on pages
- team page
- projects page
- style: alternate colors and fonts (sans-serif headings + serif body text)

Should these be merged into a new template or merged into the existing template?

Tagging for visibility and comment: @ericronne @shawnbot @jmhooper @brandocalrissian

**Comment by maya** [comment]
## Another question: 

What from the OES site can and should be abstracted into the new Federalist template? cc @ericronne

**Comment by maya** [comment]
Talked with @ericronne about this. We more or less want to keep all the elements from OES, but abstract out anything specific to OES (e.g. certain icons), and make the colors Standards-based.

Open question: should this be a theme of the uswds-jekyll template to incorporate the new styles and components or its own template? We think a theme of the template would be more flexible, but need to sort out the implementation of this. For example: how to plug in in the new style, how someone would easily plug in new sections on the homepage, so you can mix and match.

cc @shawnbot for visibility. @wslack is asking us to hold on further work and pick it up again next week.

**Comment by wslack** [comment]
Thanks Maya! One question.

> We think a theme of the template would be more flexible

Meaning an alternate option for using uswds-jekyll, right?

**Comment by maya** [comment]
@wslack I don't know how we want to implement it yet. It could be something you "switch on" from `uswds-jekyll` template, so one template + an alternate theme. Or the theme could pull in `uswds-jekyll` as a base and have the additions added to it.

**Comment by shawnbot** [comment]
I think that site-specific modifications should live in the site, but we could consider beefing up the template with some ready-to-use color palettes or document how to override only the variables that make sense. I'm not sure that we have enough use cases on which to build a general solution yet, though.

**Comment by maya** [comment]
via @shawnbot: we could control themes in the `config.yml` file and import the specified theme scss file like this in our `main.scss`:
```
---
---
@import '{{ site.theme_variant | default: 'basic' }}';
```

This sounds like the most robust and flexible solution.

**Comment by wslack** [comment]
related: https://github.com/18F/federalist/issues/887

**Comment by wslack** [comment]
Assigning to @hbillings to consider

**Comment by hursey013** [comment]
I put together a PR to implement a version of the ideas discussed above. Using the `uswds-jekyll` theme, users can set `theme_variant` in their `_config.yml` to any available variant (I've included one demo one in the PR). The variants allow a new color palette to be selected for primary, primary-alt, and secondary colors, as well as some typography customization and the ability to include additional images and stylesheets to make further customizations. 

https://github.com/18F/uswds-jekyll/pull/112

**Comment by ericronne** [comment]
@hursey013 i love this approach!

**Comment by hursey013** [comment]
Sample theme variant `beach` 🏖 :

![localhost_4000_ 5](https://user-images.githubusercontent.com/1178494/35641600-a5aa6d64-068e-11e8-8209-6bf305c6cc33.png)

**Comment by wslack** [comment]
@ericronne I think this issue scope is wider than what Brian did....should @davidewarren open a new issue with a more limited scope to reflect the work here?

**Comment by ericronne** [comment]
That might be wise. That new issue may be blocked by this one, though.

**Comment by wslack** [comment]
hhhmmmm @line47 WDYT?

**Comment by line47** [comment]
I think the scope of this work is related to the template consolidation work as well as adding additional functionality to the uswds-jekyll repo. I think this can be closed and more specific issues can be added.

**Comment by wslack** [comment]
This is about specifically augmenting our template with OES's work since it would be generally useful.
